pkg/controllers/execution/execution_controller.go

```diff
@@ -111,18 +116,46 @@ func (c *Controller) SetupWithManager(mgr controllerruntime.Manager) error {
 
 func (c *Controller) syncWork(clusterName string, work *workv1alpha1.Work) (controllerruntime.Result, error) {
 	err := c.syncToClusters(clusterName, work)
+	resource := c.tryGetResource(work)
```

Comment:
My concern is if it costs too much to add an event? We need two requests to get the resource.